wordpress-mobile/WordPress-Android
# ExPlat: Add A/B experiment

This PR is a follow up to #13960 and introduces a A/B experiment using our ExPlat client.

The experiment is pretty simple and consists of changing the position of the "Story post" action inside the menu that appears when the main FAB is tapped. 

When the variation is `Control`, the action is shown at its default position, at the bottom of the list. When the variation is `Treatment`, the action is shown at the top of the list:

| Control | Treatment |
| ------- | ----------- |
|![story-last](https://user-images.githubusercontent.com/830056/111845762-4535b200-88e4-11eb-8460-11c0ebedb77c.png)|![story-first](https://user-images.githubusercontent.com/830056/111845768-4830a280-88e4-11eb-8ce0-2d90821f2167.png)|

### To test

Notes: 
- You'll need a tool like Charles to intercept the ExPlat endpoint response.
- The ExPlat endpoint looks like this: `/wpcom/v2/experiments/0.1.0/assignments/wpandroid/`.

1. Open the app.
1. On Charles, make sure you see the normal response from the ExPlat endpoint.
1. On the My Site screen, tap the FAB.
1. Notice that the menu actions have the following order: "Blog post", "Site page" and "Story post".
1. Close the app and make sure to stop it.
1. Open the app again.
1. On Charles, intercept and edit the response so it looks like this: 
```json
{
	"variations": {
		"wpandroid_create_menu_story_first": "treatment"
	},
	"ttl": 60
}
```
8. On the My Site screen, tap the FAB.
9. Notice that the menu actions have the following order: "Story post", "Blog post", "Site page".

Here's a video demonstrating this process:

https://user-images.githubusercontent.com/830056/111847100-f2a9c500-88e6-11eb-8387-b71f40309c7f.mov

Here are my breakpoint settings on Charles:

<p float="left">
<img width="400" alt="breakpoint-settings-1" src="https://user-images.githubusercontent.com/830056/111847258-343a7000-88e7-11eb-9ca9-56db0384d25c.png">
<img width="400" alt="breakpoint-settings-2" src="https://user-images.githubusercontent.com/830056/111847260-356b9d00-88e7-11eb-8c2f-b3d7c1370601.png">
</p>








Let me know if you need additional assistance with any of the steps above:

---

PR submission checklist:
